docs/spec/spec.md

```diff
@@ -0,0 +1,297 @@
+| --- | --- | --- |
+| Create | Provisioner controller watches for Sources and creates the backing resources depending on implementation. | |
+| Update | Provisioner controller synchronizes backing implementation on changes. | |
+| Delete | Provisioner controller will deprovision backing resources depending on implementation. | |
+
+---
+
+## kind: Channel
+
+### group: eventing.knative.dev/v1alpha1
+
+_A Channel logically receives events on its input domain and forwards them to
+its subscribers. Additional behavior may be introduced by using the
+Subscription's call parameter._
+
+### Object Schema
+
+#### Spec
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| provisioner*| ProvisionerReference | The name of the provisioner to create the resources that back the Channel. | Immutable. |
+| arguments | runtime.RawExtension (JSON object)| Arguments to be passed to the provisioner. | |
+| channelable | Channelable | Holds a list of downstream subscribers for the channel. | |
+| eventTypes |[]String| An array of event types that will be passed on the Channel. | Must be objects with kind:EventType. |
+
+\*: Required
+
+#### Metadata
+
+##### Owner References
+
+ * If the Pipeline controller created this Channel: Owned by the originating Pipeline.
+ * Owned by the Provisioner used to provision the Channel.
+
+#### Status
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| Sinkable | Sinkable | Address to the endpoint as top-level domain that will distribute traffic over the provided targets from inside the cluster. | |
+| Subscribable | Subscribable | | |
+| Conditions | Conditions | Standard Subscriptions| |
+
+##### Conditions
+
+ * **Ready.** True when the Channel is provisioned and ready to accept events.
+ * **Provisioned.** True when the Channel has been provisioned by a controller.
+
+#### Events
+
+ * Provisioned
+ * Deprovisioned
+
+### Life Cycle
+
+| Action | Reactions | Limitations |
+| --- | --- | --- |
+| Create | The Provisioner referenced will take ownership of the Channel and begin provisioning the backing resources required for the Channel depending on implementation. | Only one Provisioner is allowed to be the Owner for a given Channel. |
+| Update | The Provisioner will synchronize the Channel backing resources to reflect the update. | |
+| Delete | The Provisioner will deprovision the backing resources if no longer required depending on implementation. | |
+
+---
+
+## kind: Provisioner
+
+### group: eventing.knative.dev/v1alpha1
+
+_Describes an abstract configuration of a Source system which produces events
+or a Channel system that receives and delivers events._
+
+### Object Schema
+
+#### Spec
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| type*| TypeMeta<sup>1</sup> | The type of the resource to be provisioned. | Must be Source or Channel. |
+| parameters |[]ParameterSpec| Parameters are used for validation of arguments. | |
```

Comment:
Isn't this Arguments now? Then ParameterSpec and ArugmentValueReference can be deleted.

Reply:
thanks.